### PR TITLE
DTOs + ML model api

### PR DESCRIPTION
**Pull request by geohacker**
This PR:
* [x] uses schematics
* [x] implements model subscription, deletion, and fetch API

cc @batpad

### CODE REVIEW

File: ml_enabler/api/ml.py [hunk 1]
```diff
@@ -1,7 +1,67 @@
...
+            model_id = MLModelService.subscribe_ml_model(model_dto)
+            return {"model_id": model_id}, 200
+        except DataError as e:
+            current_app.logger.error(f'Error validating request: {str(e)}')
+            return str(e), 400
+        except IntegrityError as e:
+            current_app.logger.error(f'A model with the same name already exists: {str(e)}')
+            return str(e), 400
+
+    def delete(self, model_id):
```
Comment (wronk) [inline ml_enabler/api/ml.py:73]: Can you add docstrings for this and `get` and include a few words about the parameter values

File: ml_enabler/models/ml_model.py [hunk 2]
```diff
@@ -46,17 +47,24 @@ class MLModel(db.Model):
...
-    name = db.Column(db.String)
+    name = db.Column(db.String, unique=True)
     source = db.Column(db.String)
     dockerhub_url = db.Column(db.String)
     dockerhub_hash = db.Column(db.String)
     predictions = db.relationship(Prediction, backref='ml_models',
                                   cascade='all, delete-orphan', lazy='dynamic')
 
-    def create(self):
+    def create(self, ml_model_dto: MLModelDTO):
```
Comment (wronk) [inline ml_enabler/models/ml_model.py:57]: Is this supposed to have a colon? Is requiring an explicit type as some sort of DB property? I actually don't know, haven't seen this in python.

 Can you add a few works about the 2nd param

**Review by wronk — APPROVED** [review]
Minor suggestions, but otherwise LGTM

File: ml_enabler/api/ml.py (hunk 1 above)
Comment (geohacker) [reply ml_enabler/api/ml.py:73]: Yes I'll add docstrings to all the API methods.

File: ml_enabler/models/ml_model.py (hunk 2 above)
Comment (geohacker) [reply ml_enabler/models/ml_model.py:57]: This a type hinting convention in Python https://docs.python.org/3/library/typing.html

**Pull request merged**